src/app/modules/payments/payments.service.ts

```diff
@@ -50,7 +50,9 @@ export const findPaymentById = (
       paymentId,
       null,
       // readPreference from transaction isn't respected, thus we are setting it on operation
-      session ? { session, readPreference: 'primary' } : null,
+      session
+        ? { session, readPreference: 'primary' }
```

Comment:
Suggestion: my understanding is that when the payment id is read, the suspicion is that the changes have not yet been replicated to the secondary database nodes. 

Hence, we are forcing a read to the primary node which is where the payment document is first being written to.

Perhaps we should place a comment to explain this here. 

Also, there is a caveat where this could potentially lead to increased uneven load distribution to our primary node.